src/parser/review-incentivizer-module.ts

```diff
@@ -0,0 +1,175 @@
+
+    const linkedPullNumber = (
+      await collectLinkedMergedPulls(this.context, {
+        owner: owner,
+        repo: repo,
+        issue_number: data.self?.number,
+      })
+    ).slice(-1)[0]?.number;
+
+    this.context.logger.info(`Pull request ${linkedPullNumber} is linked to this issue`);
+
+    const linkedPullReviews = await getPullRequestReviews(this.context, {
+      owner: owner,
+      repo: repo,
+      pull_number: linkedPullNumber,
+    });
+
+    for (const username of Object.keys(result)) {
+      const reward = result[username];
+      reward.reviewReward = {};
+
+      const reviewsByUser = linkedPullReviews.filter((v) => v.user?.login === username);
+
+      reward.reviewReward.reviewBaseReward = reviewsByUser.some(
+        (v) => v.state === "APPROVED" || v.state === "CHANGES_REQUESTED"
+      )
+        ? { reward: this._conclusiveReviewCredit }
+        : { reward: 0 };
+
+      const reviewDiffs = await this.fetchReviewDiffRewards(owner, repo, reviewsByUser);
+
+      reward.reviewReward.reviews = reviewDiffs;
+    }
+
+    return result;
+  }
+
+  async getTripleDotDiffAsObject(owner: string, repo: string, baseSha: string, headSha: string): Promise<CommitDiff> {
+    const response = await this.context.octokit.rest.repos.compareCommits({
+      owner,
+      repo,
+      base: baseSha,
+      head: headSha,
+    });
+
+    const files = response.data.files || [];
+    const diff: CommitDiff = {};
+
+    for (const file of files) {
+      diff[file.filename] = {
+        addition: file.additions || 0,
+        deletion: file.deletions || 0,
+      };
+    }
+
+    return diff;
+  }
+  async getReviewableDiff(
+    owner: string,
+    repo: string,
+    baseSha: string,
+    headSha: string,
+    excludedFilePatterns?: string[]
+  ) {
+    const diff = await this.getTripleDotDiffAsObject(owner, repo, baseSha, headSha);
+    const reviewEffect = { addition: 0, deletion: 0 };
+    for (const [fileName, changes] of Object.entries(diff)) {
+      if (
+        !excludedFilePatterns ||
+        (excludedFilePatterns && !excludedFilePatterns.every((pattern) => minimatch(fileName, pattern)))
+      ) {
+        reviewEffect.addition += changes.addition;
+        reviewEffect.deletion += changes.deletion;
+      }
+    }
+    return reviewEffect;
+  }
+
+  async fetchReviewDiffRewards(owner: string, repo: string, reviewsByUser: GitHubPullRequestReviewState[]) {
+    if (reviewsByUser.length == 0) {
+      return;
+    }
+    const reviews: ReviewScore[] = [];
+    const priority = parsePriorityLabel(this.context.payload.issue.labels);
+    const pullNumber = Number(reviewsByUser[0].pull_request_url.split("/").slice(-1)[0]);
+
+    const pullCommits = (
+      await this.context.octokit.rest.pulls.listCommits({
+        owner: owner,
+        repo: repo,
+        pull_number: pullNumber,
+      })
+    ).data;
+
+    // Get the commit before the PR
+    const firstCommitSha = pullCommits[0].parents[0].sha;
```

Comment:
```suggestion
    const firstCommitSha = pullCommits[0]?.parents[0]?.sha;
```
add a check if `firstCommitSha` is undefined